apps/blockchain/lib/blockchain/contract/create_contract.ex

```diff
@@ -53,22 +53,7 @@ defmodule Blockchain.Contract.CreateContract do
     contract_address = new_account_address(params)
     account = Account.get_account(original_state, contract_address)
 
-    if Account.exists?(account) do
-      cond do
-        account_will_collide?(account) ->
-          error(original_state)
-
-        account.nonce == 0 && Account.is_simple_account?(account) &&
-            not_in_contract_create_transaction?(params) ->
-          new_state =
-            increment_nonce_of_touched_account(original_state, params.config, contract_address)
-
-          {:ok, {new_state, params.available_gas, SubState.empty()}}
-
-        true ->
-          {:ok, {original_state, 0, SubState.empty()}}
-      end
-    else
+    if Account.is_simple_account?(account) do
```

Comment:
Can you a simple tests for this case? (existent account vs simple account)

Reply:
Good call! Added: [Add a testcase for when an account has a balance but no code](https://github.com/poanetwork/mana/pull/439/commits/550fc14492a63934c5c0640e8f05cc7ca586667a)